src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -105,213 +86,216 @@ private static void WriteTestDllAndName(TextWriter writer, string dll, string da
         /// <param name="level">Specifies the level of granuality to use for the output: either namepspace, class or function</param>
         /// <param name="writer">The StreamWriter which will be passed to WriteTestDllAndName() </param>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
-        public static void GetTests(IList<string> dlls, Level level, TextWriter writer)
+        public static void GetTests(IList<string> dlls, Level level)
         {
-            List<string> testsWritten = new List<string>();
-
-            List<string> categoryNames = new List<string>();
+            if (dlls == null || dlls.Count == 0)
+            {
+                throw new ArgumentNullException("dlls", "no dlls were specified");
+            }
 
-            if (dlls != null)
+            foreach (string dll in dlls)
             {
-                foreach (string dll in dlls)
-                {
-                    Assembly assembly = null;
-                    try
-                    {
-                        assembly = Assembly.LoadFrom(dll);
-                    }
-                    catch (FileNotFoundException e)
-                    {
-                        Console.Error.WriteLine("{0}: file {1} was not found on system", e.GetType(), dll);
-                        continue;
-                    }
+                List<Type> testSuites = getValidTestSuites(dll);
```

Comment:
[Could] You could use `var` instead of `List<Type>`
I usually use var for everything except basic types like int & string...  That way if you ever refactor later and the return type changes, it'll create less noise in the code review.

Reply:
This code has been changed in the next version